frontend/src/constants/DaysInWeek.ts

```diff
@@ -1,13 +1,43 @@
 import { Locale } from "date-fns";
 import { enUS } from "date-fns/locale";
 
+export const convertTime = (dateToConvert: string): string => {
+  return new Date(dateToConvert).toLocaleTimeString(navigator.language, {
+    hour: "2-digit",
+    minute: "2-digit",
+    hour12: true,
+  });
+};
+
 export const colorMap = {
   "One Time": "spinach",
   Daily: "H2O",
   Weekly: "onion",
   Monthly: "turnip",
 };
 
+export const getNextDropOff = (currentDate: string, frequency: string): string => {

```

Comment:
The problem with this approach to get the next donation is that the user can delete their next donation and then this upcoming donation will be showing wrong information. Plus I think this is the next donation for the business not the recurring donation (I asked Stacy to confirm this).

I think a better approach will be to filter the schedules that you already fetched from the backend by the donor id of the selected donation and then by the start date and pick the first one if it exists.